### DISCUSSION TITLE
How does modV compile chains of shaders?

### DISCUSSION THREAD

**Discussion by brucelane** [vcync/modV#514] [Q&A] [answered]
I will dive in the code soon, but is there a chain of shaders resulting in a shader containing the groups' effects, textures bindings, etc?

**Comment by 2xAA** [comment, marked as answer]
Hey Bruce,

So, modV doesn't compile a new shader.
Each module outputs to a texture and that is passed to the next.

This allows us to interop between different web APIs, such as webGL, webGL2 and Canvas2D.

**Comment by brucelane** [comment]
Hi Sam,

I get it, thank you
